ts/Extensions/Annotations/Popup.ts

```diff
@@ -795,162 +875,381 @@ H.Popup.prototype = {
          * (EDIT mode).
+         *
          * @private
+         *
+         * @param {Highcharts.AnnotationChart} chart
+         *        The chart object.
+         *
+         * @param {string} [optionName]
+         *        Name of the option into which selection is being added.
+         *
+         * @param {HTMLDOMElement} [parentDiv]
+         *        HTML parent element.
+         *
+         * @param {string} listType
+         *        Type of list depending on the selected bookmark.
+         *        Might be 'add' or 'edit'.
+         *
+         * @param {string|undefined} filter
+         *        Applied filter string from the input.
+         *        For the first iteration, it's an empty string.
+         *
+         * @return {void}
          */
         addIndicatorList: function (
             this: Highcharts.Popup,
             chart: Highcharts.AnnotationChart,
             parentDiv: HTMLDOMElement,
-            listType: string
+            listType: string,
+            filter?: string
         ): void {
-            let _self = this,
-                lhsCol = parentDiv.querySelectorAll('.' + PREFIX + 'popup-lhs-col')[0],
-                rhsCol = parentDiv.querySelectorAll('.' + PREFIX + 'popup-rhs-col')[0],
+            const popup = this,
+                indicators = popup.indicators,
+                lhsCol = parentDiv.querySelectorAll('.' + PREFIX + 'popup-lhs-col')[0] as HTMLElement,
+                rhsCol = parentDiv.querySelectorAll('.' + PREFIX + 'popup-rhs-col')[0] as HTMLElement,
                 isEdit = listType === 'edit',
+                addFormFields = this.indicators.addFormFields,
                 series = (
                     isEdit ?
                         chart.series : // EDIT mode
                         chart.options.plotOptions // ADD mode
-                ),
-                addFormFields = this.indicators.addFormFields,
-                rhsColWrapper: Element,
+                );
+
+            let rhsColWrapper: HTMLElement,
                 indicatorList: HTMLDOMElement,
-                item: HTMLDOMElement;
+                item: HTMLDOMElement,
+                filteredSeriesArray: [Highcharts.FilteredSeries] = [] as any;
 
-            if (!chart) {
+            if (!chart && series) {
                 return;
             }
 
-            // create wrapper for list
-            indicatorList = createElement(UL, {
-                className: PREFIX + 'indicator-list'
-            }, null as any, lhsCol as any);
+            // Filter and sort the series.
+            if (!isEdit && series && !isArray(series)) {
+                // Apply filters only for the 'add' indicator list.
+                filteredSeriesArray = indicators.filterSeries.call(this, series, filter);
+            } else if (series && isArray(series)) {
+                filteredSeriesArray = indicators.filterSeriesArray.call(this, series);
+            }
```

Comment:
Maybe sort series here? Instead of duplicating sorting code in both `filterSeries` and `filterSeriesArray` functions.